syft/format/cyclonedxjson/decoder.go

```diff
@@ -68,8 +73,7 @@ func (d decoder) Identify(reader io.ReadSeeker) (sbom.FormatID, string) {
 	dec := json.NewDecoder(reader)
 
 	var doc Document
-	err := dec.Decode(&doc)
-	if err != nil {
+	if err = dec.Decode(&doc); err != nil {
 		// maybe not json? maybe not valid? doesn't matter, we won't process it.
```

Comment:
Would it make sense to return an error here? It might be helpful to be able to surface to the UI so that users have an easier time fixing their document.

Reply:
in the decoding path that makes sense, but this is in the identification path -- there is no guarantee that what was passed in could be even JSON, so raising up errors is noisy.

Reply:
(If we make a change here it would be an update for another PR)